libs/vkd3d/command.c

```diff
@@ -618,12 +619,24 @@ static void vkd3d_wait_for_gpu_timeline_semaphore(struct vkd3d_fence_worker *wor
     vkd3d_shader_debug_ring_kick(&device->debug_ring, device, false);
     vkd3d_descriptor_debug_kick_qa_check(device->descriptor_qa_global_info);
 
-    if (fence->fence_info.fence && !is_shared_ID3D12Fence1(fence->fence_info.fence) && fence->fence_info.signal_mode)
+    if (fence->fence_info.fence && !is_shared_ID3D12Fence1(fence->fence_info.fence))
     {
         local_fence = impl_from_ID3D12Fence1(fence->fence_info.fence);
-        TRACE("Signaling fence %p value %#"PRIx64".\n", local_fence, fence->fence_info.vk_semaphore_value);
-        if (FAILED(hr = d3d12_fence_signal(local_fence, worker, fence->fence_info.vk_semaphore_value)))
-            ERR("Failed to signal D3D12 fence, hr %#x.\n", hr);
+
+        if (fence->fence_info.signal_mode == VKD3D_FENCE_SIGNAL_GPU_PHYSICAL)
+        {
+            TRACE("Signaling fence %p physical value %#"PRIx64".\n", local_fence, fence->fence_info.vk_semaphore_value);
+
+            if (FAILED(hr = d3d12_fence_signal(local_fence, worker, fence->fence_info.vk_semaphore_value)))
+                ERR("Failed to signal D3D12 fence, hr %#x.\n", hr);
+        }
+        else if (fence->fence_info.signal_mode == VKD3D_FENCE_SIGNAL_CPU_VIRTUAL)
+        {
+            TRACE("Signaling fence %p virtual value %#"PRIx64".\n", local_fence, fence->fence_info.vk_semaphore_value);
```

Comment:
Shouldn't that be fence->fence_info.virtual_value?

Reply:
derp